### PR TITLE
Add the architecture function to OSPlugin

### PR DESCRIPTION
**Pull request by pyrco**
This function is now mandatory to be implemented by all OSPlugin subclasses.

(DIS-2386)

### CODE REVIEW

File: dissect/target/plugin.py [hunk 1]
```diff
@@ -384,6 +384,18 @@ def os(self) -> str:
         """
         raise NotImplementedError
 
+    @export(property=True)
+    def architecture(self) -> Optional[str]:
+        """Required OS function.
+
+        Implementations must be decorated with ``@export(property=True)``
```
Comment (Schamper) [inline dissect/target/plugin.py:391]:
```suggestion
        Implementations must be decorated with ``@export(property=True)``.
```

File: dissect/target/plugin.py [hunk 2]
```diff
@@ -384,6 +384,18 @@ def os(self) -> str:
...
+
+        Returns:
+            A slug of the OS architecture, e.g. 'x86_32-unix', 'MIPS-linux' or
```
Comment (Schamper) [inline dissect/target/plugin.py:394]:
```suggestion
            A slug of the OS architecture, e.g. 'x86_32-unix', 'MIPS-linux' or
```
Should we standardize this to target triplets? https://wiki.osdev.org/Target_Triplet

File: dissect/target/plugins/os/unix/bsd/osx/_os.py [hunk 3]
```diff
@@ -97,3 +97,9 @@ def users(self) -> Iterator[UnixUserRecord]:
     @export(property=True)
     def os(self) -> str:
         return OperatingSystem.OSX.value
+
+    @export(property=True)
+    def architecture(self) -> str:
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/bsd/osx/_os.py:102]:
```suggestion
    def architecture(self) -> Optional[str]:
```

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/plugin.py (hunk 2 above)
Comment (pyrco) [reply dissect/target/plugin.py:394]: That would be nice yes.

File: dissect/target/plugins/general/default.py [hunk 4]
```diff
@@ -14,31 +24,35 @@ def __init__(self, target):
                 target.fs.mount(f"fs{i}", fs)
 
     @classmethod
-    def detect(cls, target):
+    def detect(cls, target: Target):
```
Comment (Schamper) [inline dissect/target/plugins/general/default.py:27]:
```suggestion
    def detect(cls, target: Target) -> Optional[Filesystem]:
```

File: dissect/target/plugins/general/default.py [hunk 5]
```diff
@@ -14,31 +24,35 @@ def __init__(self, target):
...
         pass
 
     @classmethod
-    def create(cls, target, sysvol):
+    def create(cls, target: Target, sysvol: Filesystem):
```
Comment (Schamper) [inline dissect/target/plugins/general/default.py:31]:
```suggestion
    def create(cls, target: Target, sysvol: Filesystem) -> DefaultPlugin:
```

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**